code/modules/projectiles/projectile_base.dm

```diff
@@ -80,6 +84,32 @@
 	var/shield_buster = FALSE
 	var/forced_accuracy = FALSE
 
+	///Has the projectile been fired?
+	var/has_been_fired = FALSE
+
+	//Hitscan
+	var/hitscan = FALSE //Whether this is hitscan. If it is, speed is basically ignored.
+	var/list/beam_segments //assoc list of datum/point_precise or datum/point_precise/vector, start = end. Used for hitscan effect generation.
+	/// Last turf an angle was changed in for hitscan projectiles.
+	var/turf/last_angle_set_hitscan_store
+	var/datum/point_precise/beam_index
+	var/turf/hitscan_last //last turf touched during hitscanning.
+	var/tracer_type
+	var/muzzle_type
+	var/impact_type
+
+	//Fancy hitscan lighting effects!
+	var/hitscan_light_intensity = 1.5
+	var/hitscan_light_range = 0.75
+	var/hitscan_light_color_override
+	var/muzzle_flash_intensity = 3
+	var/muzzle_flash_range = 1.5
+	var/muzzle_flash_color_override
+	var/impact_light_intensity = 3
+	var/impact_light_range = 2
+	var/impact_light_color_override
+	var/hitscan_duration = 0.3 SECONDS
```

Comment:
This is a lot of vars to add to every projectile, but I can't think of any better way besides creating something like /obj/item/projectile/hitscan. 

Also I'm just realizing, why the fuck are projectiles items?

Reply:
yeah